tests/unit/test_session.py

```diff
@@ -724,24 +723,15 @@ def unit_of_work(txn, *args, **kw):
         self.assertTrue(txn.rolled_back)
         self.assertEqual(args, ())
         self.assertEqual(kw, {})
-
-        expected_options = TransactionOptions(read_write=TransactionOptions.ReadWrite())
-        gax_api.begin_transaction.assert_called_once_with(
-            session=self.SESSION_NAME,
-            options=expected_options,
-            metadata=[("google-cloud-resource-prefix", database.name)],
-        )
-        gax_api.rollback.assert_called_once_with(
-            session=self.SESSION_NAME,
-            transaction_id=TRANSACTION_ID,
-            metadata=[("google-cloud-resource-prefix", database.name)],
-        )
+        # Transaction only has mutation operations.
+        # Exception was raised before commit, hence transaction did not begin.
```

Comment:
It's probably also worth asserting that the transaction wasn't started :+1:

Reply:
Done. Added an assertion to verify that begin transaction was not called.